### PR TITLE
Fix Issue #105 to add game information dialog

### PR DESCRIPTION
**Pull request by vickyjjj**
Added game description dialog before playing the myth fact/rapid fire games. This included editing the NewHomeActivity, adding a new res layout file for such dialogs, and adding to the strings.xml file. Attached are images of the revision.

screen shot 2016-12-03 at 10 35 52 pm
screen shot 2016-12-03 at 10 36 01 pm

### CODE REVIEW

**Comment by anubhakushwaha** [comment]
@vickyjjj Great work! Will check it.

File: malaria-app-android/src/main/java/com/peacecorps/malaria/activities/NewHomeActivity.java [hunk 1]
```diff
@@ -1,10 +1,12 @@
 package com.peacecorps.malaria.activities;
 
 import android.app.Activity;
+import android.app.Dialog;
 import android.content.Intent;
 import android.os.Bundle;
 import android.view.View;
 import android.widget.Button;
+import android.widget.TextView;
```
Comment (chhavip) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/activities/NewHomeActivity.java]: remove extra line

File: malaria-app-android/src/main/java/com/peacecorps/malaria/activities/NewHomeActivity.java [hunk 2]
```diff
@@ -92,7 +94,35 @@ public void onClick(View view) {
         return new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                startActivity(new Intent(NewHomeActivity.this, MythFactGame.class));
+                final Dialog alertDialog = new Dialog(NewHomeActivity.this,android.R.style.Theme_DeviceDefault_Dialog_NoActionBar);
+                alertDialog.setContentView(R.layout.game_info_dialog);
+                alertDialog.setCancelable(false);
+
+                // Setting Dialog Title
+                TextView title = (TextView) alertDialog.findViewById(R.id.gameTitle);
```
Comment (chhavip) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/activities/NewHomeActivity.java]: Give better variable names, such as titleTextView and so on.

File: malaria-app-android/src/main/java/com/peacecorps/malaria/activities/NewHomeActivity.java [hunk 3]
```diff
@@ -92,7 +94,35 @@ public void onClick(View view) {
...
+                TextView description = (TextView) alertDialog.findViewById(R.id.gameInfoDescription);
+                title.setText(R.string.myth_fact_game);
+                description.setText(R.string.myth_fact_description);
+                Button cancel = (Button) alertDialog.findViewById(R.id.noGameDialogButton);
```
Comment (chhavip) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/activities/NewHomeActivity.java]: cancelButton

File: malaria-app-android/src/main/java/com/peacecorps/malaria/activities/NewHomeActivity.java [hunk 4]
```diff
@@ -92,7 +94,35 @@ public void onClick(View view) {
...
+                Button ok = (Button) alertDialog.findViewById(R.id.continueGameDialogButton);
+                ok.setOnClickListener(new View.OnClickListener() {
```
Comment (chhavip) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/activities/NewHomeActivity.java]: okButton

File: malaria-app-android/src/main/java/com/peacecorps/malaria/activities/NewHomeActivity.java [hunk 5]
```diff
@@ -92,7 +94,35 @@ public void onClick(View view) {
...
+                    @Override
+                    public void onClick(View view) {
+                        alertDialog.dismiss();
+                    }
+                });
+
+                // Showing Alert Message
+                alertDialog.show();
+
+
```
Comment (chhavip) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/activities/NewHomeActivity.java]: remove extra lines

File: malaria-app-android/src/main/res/layout/game_info_dialog.xml [hunk 6]
```diff
@@ -0,0 +1,89 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="fill_parent"
+    android:layout_height="wrap_content"
+    android:background="@drawable/dialog_background"
+    android:id="@+id/dialogGameInfo">
+
```
Comment (chhavip) [inline malaria-app-android/src/main/res/layout/game_info_dialog.xml]: remove extra line

File: malaria-app-android/src/main/res/layout/game_info_dialog.xml [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+    <LinearLayout
+        android:layout_width="fill_parent"
+        android:layout_height="wrap_content"
+        android:orientation="vertical">
+
```
Comment (chhavip) [inline malaria-app-android/src/main/res/layout/game_info_dialog.xml]: same

File: malaria-app-android/src/main/res/layout/game_info_dialog.xml [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+                android:text="Start Game"
+                android:layout_marginTop="5dp"
+                android:layout_marginRight="10dp"
+                android:layout_marginLeft="5dp"
+                android:background="@drawable/info_hub_button"
+                android:textColor="@color/white"
+                />
+
+        </LinearLayout>
+
```
Comment (chhavip) [inline malaria-app-android/src/main/res/layout/game_info_dialog.xml]: same

File: malaria-app-android/src/main/res/layout/game_info_dialog.xml [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+        <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+            android:layout_width="fill_parent"
+            android:layout_height="fill_parent"
+            android:orientation="vertical"
+            android:layout_margin="30dp">
+
+            <TextView
+                android:layout_width="match_parent"
```
Comment (chhavip) [inline malaria-app-android/src/main/res/layout/game_info_dialog.xml]: ids should be at the top in a view item.

File: malaria-app-android/src/main/res/layout/game_info_dialog.xml [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+                android:layout_marginRight="10dp"
+                android:layout_marginLeft="5dp"
+                android:background="@drawable/info_hub_button"
+                android:textColor="@color/white"
+                />
+            <Button
+                android:id="@+id/continueGameDialogButton"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:text="Start Game"
```
Comment (chhavip) [inline malaria-app-android/src/main/res/layout/game_info_dialog.xml]: do not hard code strings

**Comment by vickyjjj** [comment]
@chhavip Thanks for the suggestions. Fixed.

File: malaria-app-android/src/main/java/com/peacecorps/malaria/activities/NewHomeActivity.java [hunk 11]
```diff
@@ -92,7 +94,33 @@ public void onClick(View view) {
...
+                    }
+                });
+                cancelButton.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View view) {
+                        alertDialog.dismiss();
+                    }
+                });
+
+                // Showing Alert Message
```
Comment (chhavip) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/activities/NewHomeActivity.java]: remove comment as it is obvious and remove extra blank lines.

File: malaria-app-android/src/main/java/com/peacecorps/malaria/activities/NewHomeActivity.java [hunk 12]
```diff
@@ -101,7 +129,33 @@ public void onClick(View view) {
         return new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                startActivity(new Intent(NewHomeActivity.this, RapidFireGame.class));
+                final Dialog alertDialog = new Dialog(NewHomeActivity.this,android.R.style.Theme_DeviceDefault_Dialog_NoActionBar);
+                alertDialog.setContentView(R.layout.game_info_dialog);
+                alertDialog.setCancelable(false);
+
+                // Setting Dialog Title
```
Comment (chhavip) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/activities/NewHomeActivity.java]: remove comment as it is obvious and remove extra blank lines.

File: malaria-app-android/src/main/java/com/peacecorps/malaria/activities/NewHomeActivity.java [hunk 13]
```diff
@@ -101,7 +129,33 @@ public void onClick(View view) {
...
+                    }
+                });
+                cancelButton.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View view) {
+                        alertDialog.dismiss();
+                    }
+                });
+
+                // Showing Alert Message
```
Comment (chhavip) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/activities/NewHomeActivity.java]: remove comment as it is obvious and remove extra blank lines.

**Comment by vickyjjj** [comment]
Just removed them. Thanks for the suggestions!

**Comment by yatna** [comment]
@vickyjjj Great work. However I would like to suggest a small modification. Displaying the instructions every time one opens the game can be a bit undesirable as users tend to remember the previous once . What I would suggest is that the instructions must appear only the first time a  user opens a mini game . Along with this  there can be an option  to view instructions whenever the user desires.

**Comment by yatna** [comment]
@chhavip What do you think?

**Comment by vickyjjj** [comment]
@yatna I can make that work, but the final say is up to you and @chhavip, so let me know if I should try.

**Comment by yatna** [comment]
Lets see what chhavip has to say, @ankita-kalra what would you suggest.

**Comment by chhavip** [comment]
I agree with @yatna 's suggestion, the hints should only be present the first time user opens them. Apologies for the delay in response, was travelling.

**Comment by anubhakushwaha** [comment]
@vickyjjj Willing to complete the requested changes?

**Comment by anubhakushwaha** [comment]
Closing. Fixed via #262

**Pull request closed without merging**